src/Dapr.Client/DaprClientGrpc.cs

```diff
@@ -450,6 +451,114 @@ public override async ValueTask<(TValue value, string etag)> GetStateAndETagAsyn
                 cancellationToken);
         }
 
+
+        /// <inheritdoc/>
+        public override async Task ExecuteStateTransactionAsync(
+            string storeName,
+            IReadOnlyList<StateTransactionRequest> operations,
+            Dictionary<string, string> metadata = default,
+            CancellationToken cancellationToken = default)
+        {
+            ArgumentVerifier.ThrowIfNullOrEmpty(storeName, nameof(storeName));
+            ArgumentVerifier.ThrowIfNull(operations, nameof(operations));
+            if (operations.Count == 0)
+                throw new ArgumentException($"{nameof(operations)} does not contain any elements");
+
+            await this.MakeExecuteStateTransactionCallAsync(
+                storeName,
+                operations,
+                metadata,
+                cancellationToken);
+        }
+
+        /// <inheritdoc/>
+        public override async ValueTask<bool> TryExecuteStateTransactionAsync(
```

Comment:
why this separate Try method is needed, this is basically ExecuteStateTransactionAsync with catching the RpcException

Reply:
Tried to follow the convention for SaveState:-
public override async ValueTask<bool> TrySaveStateAsync<TValue>(

Reply:
TrySaveStateAsync exists because it accepts an etag which SaveState doesn't.

Reply:
approved the PR by mistake, we should address this.

Reply:
ok will remove it